teams_accounts.go

```diff
@@ -0,0 +1,131 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"net/http"
+	"time"
+)
+
+type TeamsAccount struct {
+	GatewayID    string `json:"gateway_tag"`   // Internal teams ID
+	ProviderName string `json:"provider_name"` // Auth provider
+	CloudflareID string `json:"id"`            // cloudflare account ID
```

Comment:
is there a reason the struct field is different to the JSON tag? generally we've used a 1:1 for simplicity and reduce confusion between the API and the SDK counterpart.